apis/binding/v1alpha1/servicebinding_webhook.go

```diff
@@ -104,6 +105,10 @@ func (r *ServiceBinding) ValidateUpdate(old runtime.Object) error {
 	if err != nil {
 		log.Error(err, "Update failed")
 	}
+	if r.Spec.Application.Name != "" && r.Spec.Application.LabelSelector != nil && r.Spec.Application.LabelSelector.MatchLabels != nil {
+		log.Error(err, "name and selector MUST NOT be defined in the application reference")
```

Comment:
This check should be performed when service binding is created and updated.

Reply:
Done. 87bbe9f2d5e073d274aa92c5201c85082defc83c